src/common/encryption.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 Ben Kehoe
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from dataclasses import dataclass
+from datetime import timedelta, datetime, timezone
+import os
+from typing import Union, Optional
+
+import boto3
+import aws_encryption_sdk
+
+
+@dataclass(frozen=True)
+class EncryptionClient:
+    client: aws_encryption_sdk.EncryptionSDKClient
+    key_provider: aws_encryption_sdk.key_providers.base.MasterKeyProvider
+    cache: aws_encryption_sdk.caches.base.CryptoMaterialsCache
+    materials_manager: aws_encryption_sdk.materials_managers.base.CryptoMaterialsManager
+
+    def encrypt(
+        self,
+        plaintext: Union[str, bytes],
+        # encryption_context: Optional[dict] = None,
```

Comment:
Seems like `encryption_context` can be removed entirely now?

Reply:
Well, I'm also now wondering if we should return the headers from the encrypt/decrypt operations

Reply:
Not sure. What are you thinking we'd do with them?